### PR TITLE
fix: Streamable HTTP client ping response handling

### LINKED ISSUE #576: feature: Implement proactive HTTP client keepalive ping mechanism
## Problem Statement

When our agent performs time-intensive operations (such as web scraping, form filling, waiting for page loads, or processing large datasets) using HTTP protocol, the Go MCP client fails to send periodic ping requests to maintain the connection with the server. During long-running operations, the client remains silent and does not send any 'keep alive' messages, causing the server to interpret this as a dead connection and terminate the session, resulting in the loss of all browser context.

This issue appears to be related to client-side limitations in the Go MCP library implementation. We observe that while Claude Code properly handles connection timeout parameters sent by the Playwright MCP server and maintains the connection by regularly sending ping requests, the Go MCP library-based clients exhibit concerning behavior: they do not appear to respect server timeout parameters and seem unable to proactively send ping requests over HTTP to keep connections alive during extended operations. Attempting to work around this client limitation by sending manual 'keep alive' ping requests from application code introduces additional problems as it breaks the sequence ordering of requests in the MCP protocol.

## Verification Scenarios

To demonstrate the client-side problem and validate the solution, we present two scenarios:

Scenario 1: Go MCP Client Failure [Link to video](https://www.youtube.com/watch?v=Agn36aRqLmA)
We're using the playwright-mcp image, run it locally with HTTP transport mode, and start a MITM reverse proxy to track the traffic. When assigning a long-running task to a custom agent using the Go MCP library, the client fails to send any ping requests during the operation. The trace reveals that for the same mcp-session-id, a 404 response is received after approximately 10 seconds, indicating the server has terminated the connection due to the client's failure to send keepalive ping messages.

Scenario 2: Claude Code Client Success [Link to video](https://www.youtube.com/watch?v=NM6j__-Uc4M)
Running the playwright-mcp locally with MITM proxy monitoring, we add it as an MCP server to Claude Code. When assigning the same time-intensive task to Claude, it successfully completes the operation. The trace shows that the session remains active throughout, as Claude Code's client implementation regularly sends ping requests to the server, maintaining the connection and preventing premature termination.

## Proposed Solution

Based on our analysis, we believe this client-side issue could be addressed through a few potential approach. One would involve enhancing the Go MCP client to proactively send periodic ping requests to servers during long-running operations.

### PR DESCRIPTION
**Pull request by havasik**
## Description

Fixes #576 

Support for handling https://modelcontextprotocol.io/specification/2025-03-26/basic/utilities/ping has been added for Streamable HTTP client, when clients are created with `transport.WithContinuousListening()`.

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] MCP spec compatibility implementation
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Tests only (no functional changes)
- [ ] Other (please describe):

## Checklist

- [x] My code follows the code style of this project
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the documentation accordingly

## MCP Spec Compliance

- [x] This PR implements a feature defined in the MCP specification
- [x] Link to relevant spec section: [Link text](https://modelcontextprotocol.io/specification/2025-03-26/basic/utilities/ping)
- [x] Implementation follows the specification exactly

## Summary by CodeRabbit

* New Features
  * Server-to-client ping handling and client-side ping acknowledgments for more responsive connections.
  * SSE-compatible responses accepted to support event-stream payloads.

* Refactor
  * Replaced fixed per-request deadlines with cancellation-driven contexts for faster cancellation and improved retry behavior.

* Bug Fixes
  * Handle SSE events that carry only data (no event field).

* Tests
  * Added/extended tests for continuous listening, periodic pings/acknowledgments, SSE edge cases, and improved test synchronization.

### CODE REVIEW

File: client/transport/streamable_http_test.go [hunk 1]
```diff
@@ -554,6 +554,21 @@ func startMockStreamableWithGETSupport(getSupport bool) (string, func(), chan bo
...
+					"jsonrpc": "2.0",
+					"id":      request["id"],
+					"result":  "response received",
+				}); err != nil {
+					http.Error(w, "Failed to encode response acknowledgment", http.StatusInternalServerError)
+					return
+				}
+				return
+			}
+
```
[the remark this answers is not in the document]
Comment (ezynda3) [reply client/transport/streamable_http_test.go]: @havasik could you please address this?

[the remark this answers is not in the document]
Comment (havasik) [reply client/transport/streamable_http_test.go]: @ezynda3 just addressed above, here: https://github.com/mark3labs/mcp-go/pull/578/commits/a85fb4cfaec585561167113c1bf264d6f0fe333d - tests are successful
